rest/doc_api.go

```diff
@@ -195,11 +195,29 @@ func (h *handler) handleGetAttachment() error {
 		return base.HTTPErrorf(http.StatusNotFound, "missing attachment %s", attachmentName)
 	}
 	digest := meta["digest"].(string)
-	data, err := h.db.GetAttachment(db.AttachmentKey(digest))
+	version, ok := db.GetAttachmentVersion(meta)
+	if !ok {
+		return db.ErrAttachmentVersion
+	}
+	attachmentKey := db.MakeAttachmentKey(version, docid, digest)
+	data, err := h.db.GetAttachment(attachmentKey)
 	if err != nil {
 		return err
 	}
 
+	metaOption := h.getBoolQuery("meta")
+	if metaOption {
+		meta["key"] = attachmentKey
+		bytes, err := base.JSONMarshal(meta)
+		if err != nil {
+			base.Errorf("Error marshalling attachment metadata: %v", err)
+			return base.HTTPErrorf(http.StatusInternalServerError, "error fetching attachment metadata")
+		}
+		h.setHeader("Content-Type", "application/json")
+		h.writeRawJSONStatus(http.StatusOK, bytes)
```

Comment:
I don't think we need the manual JSON marshal/rawJSON step. The following gives the request opportunity to exit before marshalling if the client can't accept a JSON content-type, as well as handling the error cases for you.

```suggestion
		h.writeJSONStatus(http.StatusOK, meta)
```

Reply:
I see what you mean, this looks more cleaner. Manual JSON marshal/rawJSON step has been removed in the latest revision.